### PR TITLE
Fix handling of schemaless Date objects

### PR DESCRIPTION
**Pull request by sfc-gh-japatel**
For running automated Tests 
#282 
CC: @C0urante

### CODE REVIEW

File: src/test/java/com/snowflake/kafka/connector/records/ConverterTest.java
Comment (sfc-gh-japatel) [inline src/test/java/com/snowflake/kafka/connector/records/ConverterTest.java:263]: The merge gates are failing because the new code path is not executed.

Comment (C0urante) [reply src/test/java/com/snowflake/kafka/connector/records/ConverterTest.java:263]: It's executed when I run locally, and when I revert the changes to `RecordService`, the test fails. There may be a difference between my local environment and the testing environment that's contributing to this.

Comment (C0urante) [reply src/test/java/com/snowflake/kafka/connector/records/ConverterTest.java:263]: Is the testing environment using the newer version of the `connect-api` artifact?

Comment (sfc-gh-japatel) [reply src/test/java/com/snowflake/kafka/connector/records/ConverterTest.java:263]: Yes, it is using newer version, I think issue is with timestamp conversion. Hence the expected result and actual results dont match. (6940800000 vs 6930000000L)

Should we put some date format?

Comment (sfc-gh-japatel) [reply src/test/java/com/snowflake/kafka/connector/records/ConverterTest.java:263]: `Timestamp.fromLogical(schema, (java.util.Date) value));`
^ is the place where is it converting the value. May be in test we should explicitly provide Time Zone to avoid this issue or being dependent on the timeformat where the test runs.

**Comment by sfc-gh-japatel** [comment]
Tests have passed, we can push this after one approval @sfc-gh-zli

**Review by sfc-gh-zli — APPROVED** [review]
lgtm

**Pull request merged**
